google-cloud-storage/lib/google/cloud/storage/bucket.rb

```diff
@@ -1382,7 +1447,12 @@ def create_file file, path = nil, acl: nil, cache_control: nil,
         #
         #   new_file = bucket.compose [file_1, file_2], "path/to/new-file.ext"
         #
-        def compose sources, destination, acl: nil, encryption_key: nil
+        def compose sources,
```

Comment:
How do sources pass in if(_*meta)_generation?

Reply:
Thanks for catching this. I'll expose options to set the `Google::Apis::StorageV1::ComposeRequest::SourceObject::ObjectPreconditions`.

Reply:
I added a `Google::Cloud::Storage::File::Reference` value object to accept the `if_generation_match` values for sources. Please let me know what you think about this approach, including the naming, thanks.

Reply:
This discussion has been moved to #11690.